git_aggregator/repo.py

```diff
@@ -193,8 +193,44 @@ def aggregate(self):
         logger.info('End aggregation of %s', self.cwd)
 
     def init_repository(self, target_dir):
-        logger.info('Init empty git repository in %s', target_dir)
-        self.log_call(['git', 'init', target_dir])
+        """Inits the local repository
+
+        If a remote is specified as a target, it will be cloned.
+        If the target has an associated branch, and that branch exists in the
+        remote repository, the clone will be limited to that branch.
+        If there is not a valid specified target, an empty repository will be
+        initialized.
+
+        :return: True if the repository was cloned
+                 False otherwise
+        """
+        repository = None
+        for remote in self.remotes:
+            if remote["name"] == self.target["remote"]:
+                repository = remote["url"]
+                break
+        branch = self.target["branch"]
+        # If no target is defined, init an empty repository
+        if not repository:
+            logger.info('Init empty git repository in %s', target_dir)
+            self.log_call(['git', 'init', target_dir])
+            return False
+        # If a target is defined, use it as the base repository
+        logger.info(
+            'Cloning git repository %s in %s',
+            repository,
+            target_dir,
+        )
+        cmd = ('git', 'clone')
```

Comment:
```suggestion
        cmd = ('git', 'clone', '--filter=blob:none')
```

And maybe you don't even need git-autoshare.

Reply:
I read the docs but don't understand. What does that do?

Reply:
@Yajo have a look at this [article](https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/).

Reply:
We've been doing tests and indeed we're gonna use it. It's a great feature. However, it's also true that it doesn't fix all the problems we want to fix, so we'll use it in combination with git-autoshare.

Reply:
If this suggestion is applied, IMHO the PR would be complete, as the rest of using git-autoshare is doable with the wrapper script and/or can be done in a later PR.

So could we move forward with this? Thanks.

Reply:
I'd do it.

Reply:
OK, thanks.